stdlib/http/src/main/java/org/ballerinalang/net/http/BallerinaHTTPConnectorListener.java

```diff
@@ -52,17 +54,24 @@
     private final HTTPServicesRegistry httpServicesRegistry;
 
     protected final Struct endpointConfig;
+    protected final String serverName;
 
     public BallerinaHTTPConnectorListener(HTTPServicesRegistry httpServicesRegistry,
                                           Struct endpointConfig) {
         this.httpServicesRegistry = httpServicesRegistry;
         this.endpointConfig = endpointConfig;
+        if (!endpointConfig.getStringField(SERVER_NAME).isEmpty()) {
+            this.serverName = endpointConfig.getStringField(SERVER_NAME);
+        } else {
+            this.serverName = getServerName();
+        }
     }
 
     @Override
     public void onMessage(HttpCarbonMessage inboundMessage) {
         try {
             HttpResource httpResource;
+            inboundMessage.setHeader(SERVER_NAME, serverName);
```

Comment:
why do we set this header to inbound message?

Reply:
As per the current implementation, the outbound response message is created  in https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java#L340.

The response is created from Response object(https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/ballerina/http/http_response.bal#L32) But  the server endpoint configuration got from service-enpoint object(https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/ballerina/http/service_endpoint.bal#L27). When we create the outbound response, we get only the inbound message from service-endpoint module.
So I have set the server name in the inbound message.